sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -83,10 +84,14 @@ trait CreateHiveTableAsSelectBase extends DataWritingCommand {
     tableDesc: CatalogTable,
     tableExists: Boolean): DataWritingCommand
 
+  // A subclass should override this with the Class of the concrete type expected to be
+  // returned from `getWritingCommand`.
+  def writingCommandClass: Class[_]
```

Comment:
@rednaxelafx . Do we need to return the class itself? Can we simply return `String` instead of `Class[_]` because we require `name` and don't use the class actually.
```scala
- def writingCommandClass: Class[_]
+ def writingCommandClassName: String
```

Reply:
In general, we had better minimize the surface of API in terms of `parameters` and `return values`. As we know, `Class[_]` has much more information than `String`(name).

And, Yes. Since I also received the same feedbacks during working on APIs in the Spark community, I have a strong preference to avoid excessive information exposure. Is there any future plan to use that function for the other purpose?